Makefile

```diff
@@ -300,13 +300,18 @@ quickstart: $(KUSTOMIZE) manifests #EXHELP Generate the installation release man
 ##@ Docs
 
 .PHONY: crd-ref-docs
-API_REFERENCE_FILENAME := operator-controller-api-reference.md
-crd-ref-docs: $(CRD_REF_DOCS)
-	rm -f $(ROOT_DIR)/docs/refs/api/$(API_REFERENCE_FILENAME)
+OPERATOR_CONTROLLER_API_REFERENCE_FILENAME := operator-controller-api-reference.md
+CATALOGD_API_REFERENCE_FILENAME := catalogd-api-reference.md
+crd-ref-docs: tidy $(CRD_REF_DOCS)
+	$(Q)go mod vendor
+	rm -f $(ROOT_DIR)/docs/refs/api/$(OPERATOR_CONTROLLER_API_REFERENCE_FILENAME)
 	$(CRD_REF_DOCS) --source-path=$(ROOT_DIR)/api \
-	--config=$(ROOT_DIR)/config/base/crd/bases/olm.operatorframework.io_clusterextensions.yaml \
-	--renderer=markdown \
-	--output-path=$(ROOT_DIR)/docs/refs/api/$(API_REFERENCE_FILENAME)
+	--config=$(ROOT_DIR)/docs/refs/api/crd-ref-docs-gen-config.yaml \
+	--renderer=markdown --output-path=$(ROOT_DIR)/docs/refs/api/$(OPERATOR_CONTROLLER_API_REFERENCE_FILENAME);
+	rm -f $(ROOT_DIR)/docs/refs/api/$(CATALOGD_API_REFERENCE_FILENAME)
+	$(CRD_REF_DOCS) --source-path=$(ROOT_DIR)/vendor/github.com/operator-framework/catalogd/api \
```

Comment:
Are we OK with grabbing this from vendor? I feel like there has to be a better option but nothing else obvious occurred to me.

Reply:
We don't vendor our dependencies in this repo, so I think we need to do something different.

However, there is a plan afoot to make a monorepo out of operator-controller and catalogd. If there isn't an interim solution for generating docs from out-of-tree APIs, we'll be able to do this after we monorepo-ize.

Reply:
Not having thought of anything else, I've updated this to do above. Let me know if you're alright with this method. It's definitely a little clumsy though, since a local environment isn't guaranteed to have `catalogd` checked out in the right spot.

Reply:
@dtfranz What do you think about using `git` to clone the version of catalogd we depend on in `go.mod` into a temporary directory and pointing to that instead?

I saw that you were able to clone catalogd in CI to the appropriate path, but I'd like to see it work without any additional configuration on the dev side.

That being said, this probably doesn't really matter that much since we are actively planning to move to a monorepo (already seems to be good consensus on the monorepo idea, now we are hashing out the _how_).

Reply:
Thanks @everettraven ! That's a good call. I'll merge this for now and do that as an immediate follow-up. As a bonus, it would simplify the github action a little bit as well.